relayer/checkpoint.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package relayer
```

Comment:
Another thing to consider is to put this file into its own package (nested inside relayer). That way you can more finely tune which methods are available for the relayer to call on it, and which methods are just private helper functions that shouldn't be called outside of this file.

Reply:
I'm going to leave the packages as-is for now. #31 will require some refactoring of `Listener` and `ApplicationRelayer` that should result in the types becoming less coupled. At that point, all of the types in `package relayer` can likely be moved into their own packages.